yb-voyager/cmd/analyzeSchema.go

```diff
@@ -433,6 +402,51 @@ func checkStmtsUsingParser(sqlInfoArr []sqlInfo, fpath string, objType string) {
 	}
 }
 
+// Checks Whether there is a GIN index
+/*
+Following type of SQL queries are being taken care of by this function -
+	1. CREATE INDEX index_name ON table_name USING gin(column1, column2 ...)
+	2. CREATE INDEX index_name ON table_name USING gin(column1 [ASC/DESC/HASH])
+*/
+func checkGinVariations(createIndexNode *pg_query.Node_IndexStmt, sqlStmtInfo sqlInfo, fpath string) {
+	indexName := createIndexNode.IndexStmt.GetIdxname()
+	relName := createIndexNode.IndexStmt.GetRelation()
+	schemaName := relName.GetSchemaname()
+	tableName := relName.GetRelname()
+	fullyQualifiedName := lo.Ternary(schemaName != "", schemaName+"."+tableName, tableName)
+	if createIndexNode.IndexStmt.GetAccessMethod() != "gin" { // its always in lower
+		return
+	} else {
+		summaryMap["INDEX"].details[GIN_INDEX_DETAILS] = true
+	}
+	/*
+		e.g. CREATE INDEX idx_name ON public.test USING gin (data, data2);
+		stmt:{index_stmt:{idxname:"idx_name" relation:{schemaname:"public" relname:"test" inh:true relpersistence:"p"
+		location:125} access_method:"gin" index_params:{index_elem:{name:"data" ordering:SORTBY_DEFAULT nulls_ordering:SORTBY_NULLS_DEFAULT}}
+		index_params:{index_elem:{name:"data2" ordering:SORTBY_DEFAULT nulls_ordering:SORTBY_NULLS_DEFAULT}}}} stmt_location:81 stmt_len:81
+	*/
+	if len(createIndexNode.IndexStmt.GetIndexParams()) > 1 {
+		summaryMap["INDEX"].invalidCount[fmt.Sprintf("%s ON %s", indexName, fullyQualifiedName)] = true
+		reportCase(fpath, "Schema contains gin index on multi column which is not supported.",
+			"https://github.com/yugabyte/yugabyte-db/issues/7850", "", "INDEX", fmt.Sprintf("%s ON %s", indexName, fullyQualifiedName),
```

Comment:
should we use the more specific one - https://github.com/yugabyte/yugabyte-db/issues/10652 ?